storag/stream: fix stream_flash_bytes_written() return value

nvlsianpu opened the pull request:
stream_flash_bytes_written() returned number o bytes physically
written to the flash. This number might be bigger than the requested
number as is aligned to the device write-block-size.

stream_flash_bytes_written() should return number of bytes
written requested by the user.

fixes #26150

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu committed:
storag/stream: fix stream_flash_bytes_written() return value

stream_flash_bytes_written() returned number o bytes physically
written to the flash. This number might be bigger than the requested
number as is aligned to the device write-block-size.

stream_flash_bytes_written() should return number of bytes
written requested by the user.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

henrikbrixandersen reviewed (approved):
This looks like a sane approach to me. I wonder if we have in-tree code, that depends on the current behaviour.

Perhaps we should document this in the release notes?

nvlsianpu committed:
includes/storage/stream_flash: clarification on API usage

Clarify that number of bytes_written means bytes number of bytes
written belongs to the user's payload.

Added API note for emphasize that any flush write should be
the last write operation in any writes row.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

nvlsianpu commented:
> I wonder if we have in-tree code, that depends on the current behaviour.

No, we don't have.

carlescufi commented on include/storage/stream_flash.h:
```diff
@@ -105,6 +105,9 @@ size_t stream_flash_bytes_written(struct stream_flash_ctx *ctx);
  * @param data data to write
  * @param len Number of bytes to write
  * @param flush when true this forces any buffered data to be written to flash
+ *        Flush write should be the last write operation in any writes row for
+ *        given context (although it is not mandatory if total data size is
+ *        multiplier of the buffer size).
```
```suggestion
 *        A flush write should be the last write operation in a sequence of write operations for
 *        given context (although this is not mandatory if the total data size is
 *        a multiple of the buffer size).
```

nvlsianpu commented:
@henrikbrixandersen I register issue for that, so it will be listed in coming release note automatically.

jdascenzio commented:
Ok, this patch corrects more elegantly the update problem I had with canopen (#26077)! thanks
